packages/stellar-axelar-std-derive/src/into_event.rs

```diff
@@ -55,12 +55,17 @@ pub fn into_event(input: &DeriveInput) -> proc_macro2::TokenStream {
             // Parse data from Val to the corresponding types,
             // and assign them to a variable with the same name as the struct field
             // E.g. let message = Message::try_from_val(env, &data.get(0));
-            let data = Vec::<Val>::try_from_val(env, &data)
-                .unwrap_or_else(|_| {
-                    let mut vec = Vec::<Val>::new(env);
-                    vec.push_back(data);
-                    vec
-                });
+            let data = if #singleton_data {
+                Vec::<Val>::try_from_val(env, &data)
```

Comment:
why are we still trying to convert to Vec<Val>. We know it's supposed to be a singleton, so the unwrap_or_else body is what should always be done